File: religion-game/CatDog_Game_Nation_Simulation/src/Buff/SmallBuff.java
package Buff;
import Religion.Religion;

public class SmallBuff extends Buff{
	private double buff_theta[][]=new double[Religion.NUMBER_OF_RELIGION+1][Religion.NUMBER_OF_RELIGION+1];//buff_theta葵
	
	public SmallBuff(String buff_name, int duration_time){//持失切1
		super(buff_name,duration_time);
		
		for(int i=0;i<=Religion.NUMBER_OF_RELIGION;i++){
			for(int j=0;j<=Religion.NUMBER_OF_RELIGION;j++){
				buff_theta[i][j]=1;
			}
		}
	}
	
	public SmallBuff(String buff_name, int duration_time, double buff_theta[][]){//持失切2
		super(buff_name,duration_time);
		this.buff_theta=buff_theta;
	}
	
	public void modifyBuffTheta(int i, int j, double modify_to){
		buff_theta[i][j]=modify_to;
	}
	
	public double getBuffThetaAt(int i, int j){
		return buff_theta[i][j];
	}
}

File: religion-game/Assets/SmallBuff.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System;

public class SmallBuff : MonoBehaviour
{
	private readonly int number_of_religion = 4;//종교 수

	private string buff_name;//버프이름
	private int duration_time;//지속시간

	private float[,] buff_theta;//=new float[number_of_religion + 1, number_of_religion + 1];//buff_theta값

	// Use this for initialization
	void Start()
	{
		buff_theta = new float[number_of_religion + 1, number_of_religion + 1];
	}

	// Update is called once per frame
	void Update()
	{

	}

	public float getBuffThetaAt(int i, int j)
	{
		return buff_theta[i, j];
	}

	public void CountTheClock()
	{
		if(duration_time == 0){
			throw new ExpiredBuffException();
		}
		else{
			duration_time--;
		}
	}
}